sdks/java/io/solace/src/main/java/org/apache/beam/sdk/io/solace/broker/SempBasicAuthClientExecutor.java

```diff
@@ -0,0 +1,171 @@
+  }
+
+  private static String subscriptionEndpoint(String messageVpn, String queueName) {
+    return String.format("/config/msgVpns/%s/queues/%s/subscriptions", messageVpn, queueName);
+  }
+
+  BrokerResponse getQueueResponse(String queueName) throws IOException {
+    String queryUrl = getQueueEndpoint(messageVpn, queueName);
+    HttpResponse response = executeGet(new GenericUrl(baseUrl + queryUrl));
+    return BrokerResponse.fromHttpResponse(response);
+  }
+
+  BrokerResponse createQueueResponse(String queueName) throws IOException {
+    String queryUrl = createQueueEndpoint(messageVpn);
+    ImmutableMap<String, Object> params =
+        ImmutableMap.<String, Object>builder()
+            .put("accessType", "non-exclusive")
+            .put("queueName", queueName)
+            .put("owner", username)
+            .put("permission", "consume")
+            .put("ingressEnabled", true)
+            .put("egressEnabled", true)
+            .build();
+
+    HttpResponse response = executePost(new GenericUrl(baseUrl + queryUrl), params);
+    return BrokerResponse.fromHttpResponse(response);
+  }
+
+  BrokerResponse createSubscriptionResponse(String queueName, String topicName) throws IOException {
+    String queryUrl = subscriptionEndpoint(messageVpn, queueName);
+
+    ImmutableMap<String, Object> params =
+        ImmutableMap.<String, Object>builder()
+            .put("subscriptionTopic", topicName)
+            .put("queueName", queueName)
+            .build();
+    HttpResponse response = executePost(new GenericUrl(baseUrl + queryUrl), params);
+    return BrokerResponse.fromHttpResponse(response);
+  }
+
+  private HttpResponse executeGet(GenericUrl url) throws IOException {
+    HttpRequest request = requestFactory.buildGetRequest(url);
+    return execute(request);
+  }
+
+  private HttpResponse executePost(GenericUrl url, ImmutableMap<String, Object> parameters)
+      throws IOException {
+    HttpContent content = new JsonHttpContent(GsonFactory.getDefaultInstance(), parameters);
+    HttpRequest request = requestFactory.buildPostRequest(url, content);
+    return execute(request);
+  }
+
+  private HttpResponse execute(HttpRequest request) throws IOException {
+    request.setNumberOfRetries(2);
+    HttpHeaders httpHeaders = new HttpHeaders();
+    boolean authFromCookie = COOKIE_MANAGER.getCookieStore().getCookies().size() > 0;
+    if (authFromCookie) {
+      setCookiesFromCookieManager(httpHeaders);
+      request.setHeaders(httpHeaders);
+    } else {
+      httpHeaders.setBasicAuthentication(username, password);
+      request.setHeaders(httpHeaders);
+    }
+
+    HttpResponse response;
+    try {
+      response = request.execute();
+    } catch (HttpResponseException e) {
+      if (authFromCookie && e.getStatusCode() == 401) {
+        COOKIE_MANAGER.getCookieStore().removeAll();
+        // execute again without cookies to refresh the token.
+        return execute(request);
+      } else {
```

Comment:
there may be other retriable return codes? e.g. from the spec does it possible to return server side error 50X, rate limiting 429, etc. Fine to leave as is for now but good to comment here if we do want to handle them in the future what should be done

Reply:
Good point, I'll leave a comment to consider handling it in the future